pandas/tests/scalar/test_timedelta.py

```diff
@@ -255,6 +322,178 @@ def test_rfloordiv(self):
         with pytest.raises(TypeError):
             ser // td
 
+    def test_mod_timedeltalike(self):
+        # GH#19365
+        td = Timedelta(hours=37)
+
+        # Timedelta-like others
+        result = td % Timedelta(hours=6)
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(hours=1)
+
+        result = td % timedelta(minutes=60)
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(0)
+
+        result = td % pd.offsets.Hour(5)
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(hours=2)
+
+        result = td % np.timedelta64(2, 'h')
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(hours=1)
+
+        result = td % NaT
+        assert result is NaT
+
+        result = td % np.timedelta64('NaT', 'ns')
+        assert result is NaT
+
+    def test_mod_numeric(self):
+        # GH#19365
+        td = Timedelta(hours=37)
+
+        # Numeric Others
+        result = td % 2
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(0)
+
+        result = td % 1e12
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(minutes=3, seconds=20)
+
+        result = td % int(1e12)
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(minutes=3, seconds=20)
+
+    def test_mod_arraylike(self):
+        # GH#19365
+        td = Timedelta(hours=37)
+
+        # Array-like others
+        result = td % np.array([6, 5], dtype='timedelta64[h]')
+        expected = np.array([1, 2], dtype='timedelta64[h]').astype('m8[ns]')
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = td % pd.TimedeltaIndex(['6H', '5H'])
+        expected = pd.TimedeltaIndex(['1H', '2H'])
+        tm.assert_index_equal(result, expected)
+
+        result = td % np.array([2, int(1e12)], dtype='i8')
+        expected = np.array([0, Timedelta(minutes=3, seconds=20).value],
+                            dtype='m8[ns]')
+        tm.assert_numpy_array_equal(result, expected)
+
+    def test_mod_invalid(self):
+        # GH#19365
+        td = Timedelta(hours=37)
+
+        with pytest.raises(TypeError):
+            td % pd.Timestamp('2018-01-22')
+
+        with pytest.raises(TypeError):
+            td % []
+
+    def test_rmod(self):
+        # GH#19365
+        td = Timedelta(minutes=3)
+
+        result = timedelta(minutes=4) % td
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(minutes=1)
+
+        result = np.timedelta64(5, 'm') % td
+        assert isinstance(result, Timedelta)
+        assert result == Timedelta(minutes=2)
+
+        result = np.array([5, 6], dtype='m8[m]') % td
```

Comment:
see my comment above. is there a reason you are not testing TDI? (generally whenever you are testing array-like)? The array-like numerics should be with the TDI tests (if they are not in this module).

Reply:
This ties in to your comments about duplication.  In general when we have A.__op__(B) and B.__rop__(A) we can either put this test in test_A, test_B, or both.  My preference is to order these the same way as NotImplemented behavior goes.  Timedelta doesn't know anything about TDI, so test_timedelta doesn't either.  (By analogy, Series doesn't know anything about DataFrame...)